src/Native/Unix/System.Security.Cryptography.Native/pal_ssl.c

```diff
@@ -326,55 +327,56 @@ static bool GetDescriptionValue(
         }
 
         *value = valueStart;
-        valueLength = index;
+        *valueLength = index;
         return true;
     }
 
     return false;
 }
 
+#define descriptionLength 256
+
 /*
 Parses the Kx, Enc, and Mac values out of the SSL_CIPHER_description and
 maps the values to the corresponding .NET enum value.
 */
 static bool GetSslConnectionInfoFromDescription(const SSL_CIPHER* cipher,
-                                                CipherAlgorithmType& dataCipherAlg,
-                                                ExchangeAlgorithmType& keyExchangeAlg,
-                                                HashAlgorithmType& dataHashAlg,
-                                                DataHashSize& hashKeySize)
+                                                CipherAlgorithmType* dataCipherAlg,
+                                                ExchangeAlgorithmType* keyExchangeAlg,
+                                                HashAlgorithmType* dataHashAlg,
+                                                DataHashSize* hashKeySize)
 {
-    const int descriptionLength = 256;
```

Comment:
Why did you make this a #define?

Reply:
This is used in statically-bounded array declaration few lines below. In C, `const` is a type as opposed to C++ and array declaration requires a constant expression, so compiler was complaining about something related to runtime-bounded array or variable-length array from ISO C99.
This is the one of few ways of definition the contant in C:

1. `extern const int MYCONST = 256` 
2. `#define MYCONST 256`
3. `enum { MYCONST = 256}`
4.. `256`
5. `sizeof(some_construct_that_is_256_in_size)`